gobblin-tunnel/src/main/java/gobblin/tunnel/AcceptHandler.java

```diff
@@ -0,0 +1,46 @@
+package gobblin.tunnel;
+
+import java.io.IOException;
+import java.nio.channels.SelectionKey;
+import java.nio.channels.Selector;
+import java.nio.channels.ServerSocketChannel;
+import java.nio.channels.SocketChannel;
+import java.util.concurrent.Callable;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Handler for accepting connections from the client on the socket the Tunnel listens on.
+ */
+class AcceptHandler implements Callable<HandlerState> {
+  private static final Logger LOG = LoggerFactory.getLogger(Tunnel.class);
+
+  private final ServerSocketChannel server;
+  private final Selector selector;
+  private final Config config;
+
+  AcceptHandler(ServerSocketChannel server, Selector selector, Config config)
+      throws IOException {
+    this.config = config;
+    this.server = server;
+    this.selector = selector;
+
+    this.server.configureBlocking(false);
+    this.server.register(selector, SelectionKey.OP_ACCEPT, this);
```

Comment:
Given the comments in `Tunnel.java` I think this two lines can be moved to the `Tunnel.open()` method.